### DISCUSSION TITLE
How to handle overlapping Polygon borders?

### DISCUSSION THREAD

**Discussion by MobileDev500** [fleaflet/flutter_map#1951] [Q&A] [answered]
So we have an application where user can select Polygon by clicking on it, and when that happens, all the Polygons will be redrawn on the map but the selected Polygon has different `borderStrokeColor`. The problem is, when we have multiple Polygons bordering each other, the borders seem to be overlapping on the map, resulting in inconsistent behavior (only some of the borders have desired color, because others are overlapping, see screenshot). What I have tried already:

1. Adding the selected Polygon last to the list of Polygons, so it would be rendered last and therefore on top. Doesn't work, unfortunately.

2. Mathematically "moving" the coordinates of overlapping Polygons. Seems to be very complex.

It seems that Google Maps Flutter package has "Z-index" for this which can be used to specify Polygon that will always be displayed on top, but there isn't similar in this package as far as I know? Any advise would be very appreciated.
<img width="379" alt="Screenshot 2024-08-23 at 11 44 11" src="https://github.com/user-attachments/assets/053eab67-b8ea-4892-845b-64e76566165a">

**Comment by ibrierley** [comment]
Something feels wrong if option 1 isn't working, sounds like it's maybe not redrawing it or something.

Potentially I would go with a 3rd option. Create a duplicate separate poly from the selected one and always display that on top.

**Reply by MobileDev500** [reply]
How would I implement the third option? I mean, there is no Z-index so how can you specify which Polygon will always be on top?

**Comment by ibrierley** [comment, marked as answer]
Duplicate it, create a separate polygon layer and just have it added in there as the last flutter_map layer.

**Reply by MobileDev500** [reply]
This is genius, I didn't know you can add multiple layers of same type. I just tested and it works! Thank you!

**Comment by ibrierley** [comment]
(being the last poly should still do this anyway, but obviously something is amiss either with fm or your code, and bit fiddly to debug that here, so the 3rd option may get around that if nothing else to test).
